flash/video/classification/data.py

```diff
@@ -0,0 +1,285 @@
+    from pytorchvideo.data.encoded_video import EncodedVideo
+    from pytorchvideo.data.encoded_video_dataset import EncodedVideoDataset, labeled_encoded_video_dataset
+    from pytorchvideo.transforms import ApplyTransformToKey
+    from torchvision.transforms import Compose
+else:
+    ClipSampler, EncodedVideoDataset, EncodedVideo, ApplyTransformToKey = None, None, None, None
+
+_PYTORCHVIDEO_DATA = Dict[str, Union[str, torch.Tensor, int, float, List]]
+
+
+class VideoClassificationPreprocess(Preprocess):
+
+    EXTENSIONS = ("mp4", "avi")
+
+    @staticmethod
+    def default_predict_transform() -> Dict[str, 'Compose']:
+        return {
+            "per_batch_transform_on_device": Compose([
+                ApplyTransformToKey(
+                    key="video",
+                    transform=K.VideoSequential(
+                        K.Normalize(torch.tensor([0.45, 0.45, 0.45]), torch.tensor([0.225, 0.225, 0.225])),
+                        data_format="BCTHW",
+                        same_on_frame=False
+                    )
+                ),
+            ]),
+        }
+
+    def __init__(
+        self,
+        clip_sampler: 'ClipSampler',
+        video_sampler: Type[Sampler],
+        decode_audio: bool,
+        decoder: str,
+        train_transform: Optional[Dict[str, Callable]] = None,
+        val_transform: Optional[Dict[str, Callable]] = None,
+        test_transform: Optional[Dict[str, Callable]] = None,
+        predict_transform: Optional[Dict[str, Callable]] = None,
+    ):
+        # Make sure to provide your transform to the Preprocess Class
+        super().__init__(
+            train_transform, val_transform, test_transform, predict_transform or self.default_predict_transform()
+        )
+        self.clip_sampler = clip_sampler
+        self.video_sampler = video_sampler
+        self.decode_audio = decode_audio
+        self.decoder = decoder
+
+    def load_data(self, data: Any, dataset: IterableDataset) -> 'EncodedVideoDataset':
+        ds: EncodedVideoDataset = labeled_encoded_video_dataset(
+            data,
+            self.clip_sampler,
+            video_sampler=self.video_sampler,
+            decode_audio=self.decode_audio,
+            decoder=self.decoder,
+        )
+        if self.training:
+            dataset.num_classes = len(np.unique([s[1]['label'] for s in ds._labeled_videos]))
+        return ds
+
+    def predict_load_data(self, folder_or_file: str) -> List[str]:
+        if os.path.isdir(folder_or_file):
+            return [f for f in os.listdir(folder_or_file) if f.lower().endswith(self.EXTENSIONS)]
+        elif os.path.exists(folder_or_file) and folder_or_file.lower().endswith(self.EXTENSIONS):
+            return [folder_or_file]
+        raise MisconfigurationException(
+            f"The provided predict output should be a folder or a path. Found: {folder_or_file}"
+        )
+
+    def _encoded_video_to_dict(self, video) -> Dict[str, Any]:
```

Comment:
This function is pretty much the same as the encoded_video_dataset get_item(). It seems like we should just use that, right?